playground/tei_bot.py

```diff
@@ -42,95 +40,71 @@ def array_to_stone(stone_array: np.array, color: str) -> List[Stone]:
         stones = [Stone(str(x + 1), str(y + 1), color) for x, y in positions]
-
         return stones
 
     @staticmethod
-    def move_stone(stone, direction, stride=1):
+    def move_stone(stone: Stone, direction: str, stride: int = 1) -> Stone:
         x = int(stone.x)
         y = int(stone.y)
 
-        if direction == 'l':
+        if direction == "l":
             x = x - stride
-        elif direction == 'r':
+        elif direction == "r":
             x = x + stride
-        elif direction == 'u':
+        elif direction == "u":
             y = y - stride
-        elif direction == 'd':
+        elif direction == "d":
             y = y + stride
-        elif direction == 'ul':
+        elif direction == "ul":
             y = y - stride
             x = x - stride
-        elif direction == 'ur':
+        elif direction == "ur":
             y = y - stride
             x = x + stride
-        elif direction == 'dl':
+        elif direction == "dl":
             y = y + stride
             x = x - stride
-        elif direction == 'dr':
+        else:  # direction == "dr":
             y = y + stride
             x = x + stride
-
         return Stone(str(x), str(y), stone.color)
 
-    def draw_board(self):
-        stone_array = self.stone_to_array(self.log)
+    def draw_board(self, log: List[Stone]):
+        stone_array = self.stone_to_array(log)
         join_column = list(map(lambda y: " ".join([self.array_to_board(x) for i, x in enumerate(y)]), stone_array))
         join_row = "\n" + "\n".join(join_column)
         print(join_row)
 
-    def suggest_positions(self, lower=1, upper=19) -> List[Stone]:
+    def suggest_positions(self, log: List[Stone], lower: int = 1, upper: int = 19) -> List[Stone]:
         suggestions = []
-        for d in ['r', 'l', 'u', 'd', 'ul', 'ur', 'dl', 'dr']:
-            position = list(map(lambda x: self.move_stone(x, d, 1), self.log))
+        for d in ["r", "l", "u", "d", "ul", "ur", "dl", "dr"]:
+            position = list(map(lambda x: self.move_stone(x, d, 1), log))
             position = list(filter(lambda x: lower <= int(x.x) <= upper and lower <= int(x.y) <= upper, position))
             suggestions = suggestions + position
 
         suggestions = set(map(lambda x: x.x + " " + x.y, suggestions))
         suggestions = list(map(lambda x: x.split(" "), suggestions))
-        suggestions = list(map(lambda z: Stone(z[0], z[1], 's'), suggestions))
-        stone_array = self.stone_to_array(self.log)
+        suggestions = list(map(lambda z: Stone(z[0], z[1], "s"), suggestions))
+        stone_array = self.stone_to_array(log)
         suggestions = self.stone_to_array(suggestions)
         suggestions = np.clip(stone_array + suggestions, -1, 0)
         suggestions = self.array_to_stone(suggestions, "s")
-
         return suggestions
 
+    def put_stone(self, log: List[Stone]) -> Stone:
+        turn = "b" if (len(log) + 1) % 4 in [0, 1] else "w"
```

Comment:
(same comment as before)





this comment is not only for this case, but 
we already have `turn_check`(https://github.com/allmeat/connect6/blob/master/playground/referee.py#L27)

so it might be better to utilize this function and standardize the way we check turn (or colors).